### PR TITLE
Issue #12441: Resolve Pitest suppression for RightCurlyCheck

### PR DESCRIPTION
**Pull request by arinmodi**
Issue #12441: Resolve Pitest suppression for RightCurlyCheck

Diff Regression config: https://gist.githubusercontent.com/arinmodi/3cf085f17de7e31a5089c43182f45eec/raw/37bb6fa0f32abfc98f35ebad85ec603fb89e665e/config2.xml

### CODE REVIEW

**Comment by arinmodi** [comment]
Github, generate report

**Review by nrmancuso — CHANGES_REQUESTED** [review]
https://github.com/checkstyle/checkstyle/issues/12441#issuecomment-1324211173

**Comment by arinmodi** [comment]
Github, generate report

**Comment by arinmodi** [comment]
Github, generate report

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java [hunk 1]
```diff
@@ -105,6 +105,7 @@ public void testAlone() throws Exception {
             "184:9: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 9),
             "189:9: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 9),
             "190:53: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 53),
+            "221:12: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 12),
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java:108]: @arinmodi please create new input file; we are trying to keep input files shorter than 120 lines: https://github.com/checkstyle/checkstyle/issues/11163. 

To test input, please add some more input form the report, such as https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/1cb9b86_2022085214/reports/diff/sevntu-checkstyle/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/sevntu-checkstyle/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/coding/InputAvoidHidingCauseExceptionCheck.java.html#L71,  and a bunch from https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/1cb9b86_2022085214/reports/diff/sevntu-checkstyle/index.html#A54

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java (hunk 1 above)
Comment (arinmodi) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+RightCurly
+option = (default)SAME
+tokens = (default)LITERAL_TRY , LITERAL_CATCH , LITERAL_FINALLY , LITERAL_IF , LITERAL_ELSE
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyTestDefault {
```
Comment (nrmancuso) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java:11]: @arinmodi do we require two inputs? If not, let's have only one please.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java (hunk 2 above)
Comment (arinmodi) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java:11]: Both the input file having different configuration that's why

Comment (nrmancuso) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java:11]: Mutation was killed with a single config in https://github.com/checkstyle/checkstyle/commit/e5b77a6fe9c2724b2215f4cadca5bbda7270aacd, why do we need two now?

Comment (arinmodi) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java:11]: You told me to add this type of input in previous item

Now, Are you sure want to remove one input file ?

Comment (nrmancuso) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestDefault.java:11]: Input code and config we check it with are different concepts; please consolidate all input to one file with config to kill mutation.

**Review by romani — APPROVED** [review]

**Review by nrmancuso — APPROVED** [review]
Good to merge, but we need green CI first. @arinmodi please rebase on latest master

**Pull request merged**